sql/catalyst/src/main/scala/org/apache/spark/sql/types/MapType.scala

```diff
@@ -73,6 +74,90 @@ case class MapType(
   override private[spark] def existsRecursively(f: (DataType) => Boolean): Boolean = {
     f(this) || keyType.existsRecursively(f) || valueType.existsRecursively(f)
   }
+
+  private[this] class OrderedWrapper {
+    var isOrdered: Boolean = false
```

Comment:
I prefer not to make this mutable if we can. That can be a source of some pretty weird errors if we move from an unordered to an ordered map. Why do you need this?

Reply:
Thanks for quick reply :)

Actually I'm not pretty sure about this.
If we do it like below
```
case class MapType(
  keyType: DataType,
  valueType: DataType,
  valueContainsNull: Boolean,
  ordered: Boolean)
```
The `ordered` will be spread to lots places in the code (especially in the `...match ... case ...` ) and users can will also  see it. But I think `ordered` is a pretty internal parameter/characteristic and only used when sorting map. So I try to make it private and lazy created.